File: src/main/java/pl/com/employeemanager/service/EmployeeService.java
package pl.com.employeemanager.service;

import pl.com.employeemanager.dto.EmployeeBaseDTO;
import pl.com.employeemanager.dto.EmployeeCriteriaDTO;
import pl.com.employeemanager.dto.EmployeeFullDTO;
import pl.com.employeemanager.model.Employee;

import java.util.List;

public interface EmployeeService {
    List<Employee> getEmployees();
    List<EmployeeBaseDTO> getEmployees(EmployeeCriteriaDTO criteria);
    EmployeeFullDTO getEmployee(Integer personId);
    Integer saveEmployee(EmployeeFullDTO employeeFullDTO);
    Integer updateEmployee(EmployeeFullDTO employeeFullDTO);
}

File: src/main/resources/dev/app/employeeList/employee.service.js
angular
    .module('em.employee')
    .service('employeeService', employeeService);

function employeeService($http) {
    var vm = this;
    vm.getEmployees = getEmployees;
    vm.getEmployeeById = getEmployeeById;
    vm.setPersonId = setPersonId;
    vm.getPersonId = getPersonId;
    vm.saveEmployee = saveEmployee;

    vm.personId = null;

    function setPersonId(personId) {
        vm.personId = personId;
    }

    function getPersonId() {
        return vm.personId;
    }

    function getEmployees(crit) {
        return $http.post('/employee-manager/employee', crit);
    }

    function getEmployeeById(personId) {
        return $http.get('/employee-manager/employee/' + personId);
    }

    function saveEmployee(employee) {
        return $http.post('/employee-manager/saveEmployee', employee);
    }
}